Framework/Crystal/src/FindSXPeaks.cpp

```diff
@@ -84,16 +61,134 @@ void FindSXPeaks::init() {
   declareProperty("EndWorkspaceIndex", EMPTY_INT(), mustBePositive,
                   "End spectrum number  (default FindSXPeaks)");
-  declareProperty("SignalBackground", 10.0,
-                  "Multiplication factor for the signal background");
+
+  // ---------------------------------------------------------------
+  // Peak strategies + Threshold
+  // ---------------------------------------------------------------
+  std::vector<std::string> peakFindingStrategy = {strongestPeakStrategy,
+                                                  allPeaksStrategy};
+  declareProperty(
+      "PeakFindingStrategy", strongestPeakStrategy,
+      boost::make_shared<StringListValidator>(peakFindingStrategy),
+      "Different options for peak finding."
+      "1. StrongestPeakOnly: Looks only for the the strongest peak in each "
+      "spectrum (provided there is "
+      "one). This options is more performant than the AllPeaks option.\n"
+      "2. AllPeaks: This strategy will find all peaks in each "
+      "spectrum. This is slower than StrongestPeakOnly.\n");
+
+  // Declare
+  declareProperty(
+      "SignalBackground", 10.0,
+      "Multiplication factor for the signal background. Peaks which are"
+      " below the estimated background are discarded. The background is "
+      "estimated"
+      " to be an average of the first and the last signal and multiplied"
+      " by the SignalBackground property.\n");
+
+  declareProperty(
+      "AbsoluteBackground", 10.0,
+      "Peaks which are below the specified absolute background are discarded."
+      " The background is gloabally specified for all spectra. Inspect your "
+      "data in the InstrumentView to get a good feeling for the background "
+      "threshold.\n"
+      "Background thresholds which are too low will mistake noise for peaks.");
+
+  // Enable
+  setPropertySettings("SignalBackground",
+                      make_unique<EnabledWhenProperty>(
+                          "PeakFindingStrategy",
+                          Mantid::Kernel::ePropertyCriterion::IS_EQUAL_TO,
+                          strongestPeakStrategy));
+
+  setPropertySettings("AbsoluteBackground",
+                      make_unique<EnabledWhenProperty>(
+                          "PeakFindingStrategy",
+                          Mantid::Kernel::ePropertyCriterion::IS_EQUAL_TO,
+                          allPeaksStrategy));
+
+  // Group
+  const std::string peakGroup = "Peak Finding Settings";
+  setPropertyGroup("PeakFindingStrategy", peakGroup);
+  setPropertyGroup("SignalBackground", peakGroup);
+  setPropertyGroup("AbsoluteBackground", peakGroup);
+
+  // ---------------------------------------------------------------
+  // Resolution
+  // ---------------------------------------------------------------
+  // Declare
+  std::vector<std::string> resolutionStrategy = {
+      relativeResolutionStrategy, absoluteResolutionPeaksStrategy};
+  declareProperty("ResolutionStrategy", relativeResolutionStrategy,
```

Comment:
1. It should depend on your data set and what you use for  `TofResolution`, `PhiResolution` and `TwoThetaResolution`. I would be surprised if turning these parameters does not result in any changes.  

2. The `AllPeaks` strategy did give the best results  with `AbsoluteResolutionStrategy`, but if a user is more comfortable with the relative approach (especially since it has been in there for years), I think we should allow them to use the `RelativeResolutionStrategy`. What do you think?